client/src/main/java/io/pravega/client/stream/impl/EventStreamReaderImpl.java

```diff
@@ -154,8 +168,18 @@
         } 
         lastRead = Sequence.create(segment.getSegmentId(), offset);
         int length = buffer.remaining() + WireCommands.TYPE_PLUS_LENGTH_SIZE;
-        return new EventReadImpl<>(deserializer.deserialize(buffer), getPosition(),
-                                   new EventPointerImpl(segment, offset, length), null);
+        addSegmentOffsetUpdateIfNeeded(segment, offset + length);
+        return new EventReadImpl<>(deserializer.deserialize(buffer),
+                PositionImpl.builder().ownedSegments(ownedSegments).segmentRanges(ranges).updatesToSegmentOffsets(segmentOffsetUpdates).build(),
```

Comment:
the maps need to be wrapped with `java.util.Collections#unmodifiableMap` and the updates should be wrapped with `Collections.unmodifiableList()`.

Reply:
Thanks, I have done this change.

Reply:
@shrids I just moved the immutability of these collections to the PositionImpl builder itself, so it takes effect to all the usages of such builder.